tests/integration/test_outcome.py

```diff
@@ -276,3 +287,94 @@ def test_outcome_source(relay, mini_sentry):
     outcome = outcomes[0]
 
     assert outcome.get("source") == "my-layer"
+
+
+def test_outcome_forwarding(relay, mini_sentry):
```

Comment:
For testing all these cases, you can just use `relay_chain`, like here:

https://github.com/getsentry/relay/blob/fea4f20c8be5b6a54153cde358bf901d3df6c7ef/tests/integration/test_envelope.py#L4-L5

This is a parameterized fixture that will create multiple tests with variable length chains and also HA proxies between them.

Reply:
yes I know the relay_chain... I wanted to explicitly control what relays I use and control their configuration so I can figure out which one is the source of the outcome